python-client/giskard/core/suite.py

```diff
@@ -162,16 +163,7 @@ class TestPartial:
 
 
 def single_binary_result(test_results: List):
-    passed = True
-    for r in test_results:
-        if type(r) == bool:
-            passed = passed and r
-        elif hasattr(r, "passed"):
-            passed = passed and r.passed
-        else:
-            logger.error(f"Invalid test result: {r.__class__.__name__}")
-            passed = False
-    return passed
+    return all(map(lambda res: res.passed, test_results))
```

Comment:
```py
not any(not res.passed for res in test_results)
```
will stop at the first failed test.

Reply:
Let’s add a unit test for this. Can mock the rest result objects.

Reply:
> ```python
> not any(not res.passed for res in test_results)
> ```
> 
> will stop at the first failed test.

Shouldn't `all` also stop at the first false? 
Something like this: `all(res.passed for res in test_results)`